File: lib-universalAdapter/src/main/java/com/llj/adapter/UniversalConverter.java
package com.llj.adapter;

import androidx.annotation.NonNull;

/**
 * PROJECT:UniversalAdapter
 * DESCRIBE:
 * Created by llj on 2017/1/15.
 */

public interface UniversalConverter<Item, Holder extends ViewHolder> {

    void setAdapter(@NonNull UniversalAdapter<Item, Holder> listAdapter);

    UniversalAdapter<Item, Holder> getAdapter();

    void cleanup();
}

File: lib-universalAdapter/src/main/java/com/llj/adapter/UniversalConverter.kt
package com.llj.adapter

/**
 * describe 适配器基类
 *
 * @author liulinjie
 * @date 2020/4/18 12:20 AM
 */
interface UniversalConverter<Item, Holder : XViewHolder> {
    fun setAdapter(listAdapter: UniversalAdapter<Item, Holder>)
    fun getAdapter(): UniversalAdapter<Item, Holder>
    fun cleanup()
}